### PR TITLE
{CI} Generate SBOM manifest for built packages

### PR DESCRIPTION
**Pull request by wangzelin007**
**Description**

Add [ado-sbom-generator-task](https://eng.ms/docs/cloud-ai-platform/devdiv/one-engineering-system-1es/1es-docs/secure-supply-chain/ado-sbom-generator) to support Generate SBOM (Software Bill of Materials) manifest for built packages.

We need generate SBOM manifest in every artifacts directory.

image

But there are two remaining problems：
1. Required packages like azure-mgmt-resource are missing from the built artifact. SBOM team use [component-detection](https://github.com/microsoft/component-detection/blob/main/src/Microsoft.ComponentDetection.Detectors/pip/PipComponentDetector.cs#L18) to detect python packages.
Open two issues in component-detection repo.
[Component detection only find the setup.py file in tools directory.](https://github.com/microsoft/component-detection/issues/73)
[Component detection only match setup.py and requirement.txt file.](https://github.com/microsoft/component-detection/issues/74)
3. Pipeline fail when variables `Packaging.EnableSBOMSigning: true` is set.
error message: 
```diff
- ##[error]The signing feature is not available for your organization yet.
```
**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: `az command a`: Make some customer-facing breaking change.
[Component Name 2] `az command b`: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
CI

File: azure-pipelines.yml [hunk 1]
```diff
@@ -322,13 +331,18 @@ jobs:
...
+    inputs:
+      BuildDropPath: $(Build.ArtifactStagingDirectory)
 
   - task: PublishPipelineArtifact@0
     displayName: 'Publish Artifact: pypi'
     inputs:
       TargetPath: $(Build.ArtifactStagingDirectory)
       ArtifactName: pypi
 
+
```
Comment (wangzelin007) [inline azure-pipelines.yml:345]:
```suggestion
```

File: azure-pipelines.yml [hunk 2]
```diff
@@ -312,7 +322,6 @@ jobs:
     inputs:
       versionSpec: 3.10
```
Comment (wangzelin007) [inline azure-pipelines.yml:324]:
```suggestion


```

**Review by jiasli — APPROVED** [review]

**Comment by jiasli** [comment]
PR title can be "Generate SBOM manifest for built packages".

BTW if anyone is interested, SBOM stands for "Software Bill of Materials".

File: azure-pipelines.yml [hunk 3]
```diff
@@ -172,6 +172,11 @@ jobs:
        build_scripts/windows/scripts/build.cmd
     displayName: 'Build Windows MSI'
 
+  - task: AzureArtifacts.manifest-generator-task.manifest-generator-task.ManifestGeneratorTask@0
+    displayName: 'SBOM'
+    inputs:
+      BuildDropPath: 'build_scripts/windows/out/'
```
Comment (jsntcy) [inline azure-pipelines.yml:178]: Can we use `$(Build.ArtifactStagingDirectory)` for `BuildDropPath` to be consistent across all artifacts?

Comment (wangzelin007) [reply azure-pipelines.yml:178]: Sorry, we can't.
Accroding to the [doc](https://eng.ms/docs/cloud-ai-platform/devdiv/one-engineering-system-1es/1es-docs/secure-supply-chain/ado-sbom-generator), we should use the same root path location that you publish your build drop to.
For MSI, our publish path is `build_scripts/windows/out/`.
If we want to make all the artifacts BuildDropPath same, we need change a lot of scripts and pipelines.
image

**Review by jsntcy — APPROVED** [review]

**Pull request merged**